### ISSUE TITLE
Doesn't work with Scala 2.12

### ISSUE THREAD

**Issue by eed3si9n** [hamnis/dataclass-scalafix#4]
## steps

Use Scala 2.12

```scala
package lmcoursier.definitions

import dataclass._

@data class Strict(
  include: Set[(String, String)] = Set(("*", "*")),
  exclude: Set[(String, String)] = Set.empty,
) {
}
```

## problems

```scala
[error] /Users/xxxx/work/sbt-coursier/modules/lm-coursier/target/scala-2.12/src_managed/main/scala/lmcoursier/definitions/Strict.scala:37:14: method productElementName overrides nothing
[error] override def productElementName(n: Int) = n match {
[error]              ^
[error] /Users/xxxx/work/sbt-coursier/modules/lm-coursier/target/scala-2.12/src_managed/main/scala/lmcoursier/definitions/Strict.scala:51:14: method productElementNames overrides nothing
[error] override def productElementNames = {
[error]              ^
```

**Comment by hamnis** [comment]
Fixed by 0fc737007b5d82d2f7ef26c8d3a3697691636c0b

**Comment by bjaglin** [comment]
Since there is currently no mapping between the runtime the rule runs with and the target (as we don't have scalafix cross-compiled for Scala 3 just yet), to know which code you should produce, you should check `scalaVersion` from [`scalafix.v1.Configuration`](https://www.javadoc.io/static/ch.epfl.scala/scalafix-core_2.12/0.10.1/scalafix/v1/Rule.html#withConfiguration(config:scalafix.v1.Configuration):metaconfig.Configured[scalafix.v1.Rule]) instead of relying on the runtime scalaVersion.
